clusterloader2/pkg/test/simpletestexecutor.go

```diff
@@ -62,12 +62,22 @@ func (ste *simpleTestExecutor) ExecuteTest(ctx Context, conf *api.Config) []erro
 
 // ExecuteStep executes single test step based on provided step configuration.
 func (ste *simpleTestExecutor) ExecuteStep(ctx Context, step *api.Step) []error {
+	var wg wait.Group
+	var lock sync.Mutex
 	var errList []error
 	if len(step.Measurements) > 0 {
-		// TODO: handle measurements
+		for i := range step.Measurements {
+			// index is created to make i value unchangeable during thread execution.
+			index := i
+			wg.Start(func() {
+				if err := ctx.GetMeasurementManager().Execute(step.Measurements[index].Method, step.Measurements[index].Identifier, step.Measurements[index].Params); err != nil {
+					lock.Lock()
+					defer lock.Unlock()
```

Comment:
Maybe in a separate PR, but I think it makes sense to move this errList+lock combination into a struct (since you seem to use it at quite some places).

Reply:
TODO added (line 66).